### ISSUE TITLE
PaDEL needs updating to CDK 2.0

### ISSUE THREAD

**Issue by egonw** [kmansouri/OPERA#1]
We should see if we can find a few people to upgrade PaDEL to CDK 2.0.... cc @johnmay @rajarshi

**Comment by rajarshi** [comment]
Where are the PaDEL sources?

**Comment by egonw** [comment]
http://www.yapcwsoft.com/dd/padeldescriptor/PaDEL-Descriptor_src.zip

**Comment by johnmay** [comment]
Technically not a project issue... :-)

**Comment by johnmay** [comment]
Scrap that thought it was on CDK/issues 👍

**Comment by kmansouri** [comment]
Yes I'm working on integrating CDK2 in the next version. Thank you all!

**Comment by kmansouri** [comment]
I am working on using CDK2 in the next version of OPERA but I'm facing some problems...
@egonw @rajarshi @tsufz
1) Not all PaDEL descriptors (and fingerprints) are available in CDK2
So I think the solution is to upgrade PaDEL to use CDK2 (not to replace it). Has anyone figured that out? 

2) PaDEL is using CDK 1.5 which is giving different values than CDK2 for certain descriptors...
Does that mean those descriptors had issues in CDK 1.5?

**Comment by rajarshi** [comment]
1. PaDEL may have descriptors that are not in the CDK
2. There have been some bug fixes in descriptor implementations that lead to differences in computed values

**Comment by kmansouri** [comment]
Thanks, @rajarshi for the quick response! 

I'mo wondering if anyone working on updating PaDEL to use CDK2?

**Comment by rajarshi** [comment]
That'd be a question for the PaDEL devs

**Comment by kmansouri** [comment]
That makes sense :)
Any plans to include the extra PaDEL descriptors in CDK?

**Comment by ChemConnector** [comment]
Just out of interest you might be interested in the Mordred Descriptors
that discuss advantages over PaDEL
https://jcheminf.springeropen.com/articles/10.1186/s13321-018-0258-y

Tony Williams

Phone: +1 (919) 201-1516
Email: tony27587@gmail.com
Skype: tony27587
My Blog: www.chemconnector.com
Twitter: http://twitter.com/ChemConnector

**Comment by kmansouri** [comment]
Thanks a lot Tony! I'm looking into it :)

**Comment by tsufz** [comment]
It PaDEL further developed? The webpage is available again, but quite outdated (last updated: 2014). The link is: http://www.yapcwsoft.com/dd/padeldescriptor. The source code could be downloaded here:
http://www.yapcwsoft.com/dd/padeldescriptor/PaDEL-Descriptor_src.zip. However, there is no Github or other public repository.

**Comment by kmansouri** [comment]
I don't think that PaDEL will be maintained by its original developer anymore. But it'll be useful if someone else can...

**Comment by tsufz** [comment]
@egonw, maybe something to be integrated in CDK?

**Comment by egonw** [comment]
We integrated some of the PaDEL descriptors in the past... integrating more of them is welcome. Check the 5 patches in this branch: https://github.com/cdk/cdk/tree/dev-1.4.x-padel

**Comment by rajarshi** [comment]
@marcabrus a lot of those in that list appear to be duplicates. A few descriptor types that would be useful include

* Grid based descriptors derived from MIF's
* Conformational descriptors (e.g., 4D descriptors from Hopfinger and colleagues)
* Task specific descriptors (e.g., https://t.co/Zheeo7tn61)

**Comment by kmansouri** [comment]
@marcabrus 

> @kmansouri Are you still using cdk libraries 1.4.9 or you managed to update them?

The new models from OPERA 2.0 (CERAPP, CoMPARA and CATMoS) are using CDK2. The older models from previous versions are still using CDK1.48.
The goal is to migrate to CDK2 for new models or if existing models are significantly updated (like adding new data).

**Comment by marcabrus** [comment]
> @marcabrus a lot of those in that list appear to be duplicates. A few descriptor types that would be useful include
> 
> * Grid based descriptors derived from MIF's
> * Conformational descriptors (e.g., 4D descriptors from Hopfinger and colleagues)
> * Task specific descriptors (e.g., https://t.co/Zheeo7tn61)

I'm aware that lot of that descriptors are duplicates, or even diffrent names with the same meaning. 
For this reason i was referring to this map
![2019-01-22](https://user-images.githubusercontent.com/335108/51553450-2c9bef80-1e73-11e9-9304-519d96e0fda3.png)

There is a significant overlap among the various free software, but also there is a large part of decriptors not implemented in Vega.
Furthermore I need to mention that several descriptors contained in Dragon can be easily implemented.

My current resources and skills , let me to write (or adapt) code only for 2d based descriptors, graph or smarts based, nonetheless they are found relevant to several QSAR models we implemented.

The descriptors you suggested need some experience in 3D optimization, and the work on structures with a clear knowledge of isomerism. The kind of data we often analyze doesn't let us to work in this way.

These are important limitations, but maybe in the future we can overcome them.

**Comment by kmansouri** [comment]
@egonw @rajarshi @marcabrus @tsufz  anyone has updates about updating PaDEL to use CDK2? 
@marcabrus you mentioned reimplementing DRAGON/PaDEL 2D descriptors, any updates about that?
Thanks!

**Comment by tsufz** [comment]
@kmansouri, No, I did not work on any update.
